### PR TITLE
Elm ui

### PR DESCRIPTION
**Pull request by ashishnegi**
1. Currently sends data to echo server `ws://echo.websocket.org`. Just need to change that string.
2. Has many code/view to simulate server data. Would be removed once connected with tronkell server.

screen shot 2016-09-05 at 21 17 27

### CODE REVIEW

File: ui/src/Grid/Message.elm [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module Grid.Message exposing (..)
+
+import Grid.Player exposing (..)
+
+type Msg = PlayerMoved PlayerId Position Orientation
+         | PlayerDied PlayerId
+         | NoOp
```
Comment (ford-prefect) [inline ui/src/Grid/Message.elm]: This looks like we're duplicating messages from the server?

Comment (ford-prefect) [inline ui/src/Grid/Message.elm]: Also what's the function of the NoOp?

Comment (ashishnegi) [inline ui/src/Grid/Message.elm]: client would have the same message types that of server .. right ?
`NoOp` is Msg type .. i see that people use it whenever they have unnecessary conditions..
as `view` or standard functions have to have to return `Html Msg`

Comment (ford-prefect) [inline ui/src/Grid/Message.elm]: Why not share that message structure between the server and client then? Same problem of it not being a proper subset of all messages?

Comment (ashishnegi) [inline ui/src/Grid/Message.elm]: this is `elm` code. How do we share `haskell` code here ? or you are talking about something else ?
Also, not all msgs are required in client side like `GameEnded`..

Comment (ford-prefect) [inline ui/src/Grid/Message.elm]: /o\ d'oh, I should've realised you can't reuse modules between the two. Never mind my comments, then.

**Pull request closed without merging**